File: src/primes/primes.js
var intervalList=function(m, n){
    if (m > n)
        return [];
    else 
        return [m].concat(intervalList(m+1,n));
};
var sieve =function(l){
    var removeMultiples=function(n, l){
        if (l.length==0)
            return [];
        else if (0==l[0]%n)
            return removeMultiples(n, l.slice(1))
        else return [l[0]].concat(removeMultiples(n,l.slice(1))); 
    };
    if(l.length==0)
        return [];
    else return [l[0]].concat(sieve(removeMultiples(l[0],l.slice(1))));
};

var primesNotBiggerThan=function(n){
    return sieve(intervalList(2, n))
};
primesNotBiggerThan(6000);

File: src/primes/primes.py
def intervalList(m, n):
    if m > n:
        return []
    else:
        return [m] + intervalList(m + 1, n)

def sieve(l):
    def removeMultiples(n, l):
        if len(l) == 0:
            return []
        elif l[0] % n == 0:
            return removeMultiples(n, l[1:])
        else:
            return [l[0]] + removeMultiples(n, l[1:])

    if len(l) == 0:
        return []
    else:
        return [l[0]] + sieve(removeMultiples(l[0], l[1:]))

def primesNotBiggerThan(n):
    return sieve(intervalList(2, n))

primesNotBiggerThan(6000)
